skbio/io/format/genbank.py

```diff
@@ -0,0 +1,865 @@
+      ORGANISM  synthetic construct
+                other sequences; artificial sequences.
+    REFERENCE   1  (bases 1 to 34)
+      AUTHORS   Klein,D.J., Edwards,T.E. and Ferre-D'Amare,A.R.
+      TITLE     Cocrystal structure of a class I preQ1 riboswitch
+      JOURNAL   Nat. Struct. Mol. Biol. 16 (3), 343-344 (2009)
+       PUBMED   19234468
+    COMMENT     SEQRES.
+    FEATURES             Location/Qualifiers
+         source          1..34
+                         /organism="synthetic construct"
+                         /mol_type="other RNA"
+                         /db_xref="taxon:32630"
+    ORIGIN
+            1 agaggttcta gcacatccct ctataaaaaa ctaa
+    //
+
+>>> gb = ['LOCUS       3K1V_A     34 bp   RNA    linear   SYN 10-OCT-2012\\n',
+...       'DEFINITION  Chain A, Structure Of A Mutant Class-I Preq1.\\n',
+...       'ACCESSION   3K1V_A\\n',
+...       'VERSION     3K1V_A  GI:260656459\\n',
+...       'KEYWORDS    .\\n',
+...       'SOURCE      synthetic construct\\n',
+...       '  ORGANISM  synthetic construct\\n',
+...       '            other sequences; artificial sequences.\\n',
+...       'REFERENCE   1  (bases 1 to 34)\\n',
+...       "  AUTHORS   Klein,D.J., Edwards,T.E. and Ferre-D'Amare,A.R.\\n",
+...       '  TITLE     Cocrystal structure of a class I preQ1 riboswitch\\n',
+...       '  JOURNAL   Nat. Struct. Mol. Biol. 16 (3), 343-344 (2009)\\n',
+...       '   PUBMED   19234468\\n',
+...       'COMMENT     SEQRES.\\n',
+...       'FEATURES             Location/Qualifiers\\n',
+...       '     source          1..34\\n',
+...       '                     /organism="synthetic construct"\\n',
+...       '                     /mol_type="other RNA"\\n',
+...       '                     /db_xref="taxon:32630"\\n',
+...       'ORIGIN\\n',
+...       '        1 agaggttcta gcacatccct ctataaaaaa ctaa\\n',
+...       '//\\n']
+
+Now we can read it as ``DNA`` object:
+
+>>> from skbio import DNA, RNA, Sequence
+>>> dna_seq = DNA.read(gb)
+>>> dna_seq
+DNA
+-----------------------------------------------------------------
+Metadata:
+    'ACCESSION': '3K1V_A'
+    'COMMENT': 'SEQRES.'
+    'DEFINITION': 'Chain A, Structure Of A Mutant Class-I Preq1.'
+    'FEATURES': <class 'list'>
+    'KEYWORDS': '.'
+    'LOCUS': <class 'dict'>
+    'REFERENCE': <class 'list'>
+    'SOURCE': <class 'dict'>
+    'VERSION': '3K1V_A  GI:260656459'
+Positional metadata:
+    0: <dtype: bool>
+Stats:
+    length: 34
+    has gaps: False
+    has degenerates: False
+    has non-degenerates: True
+    GC-content: 35.29%
+-----------------------------------------------------------------
+0 AGAGGTTCTA GCACATCCCT CTATAAAAAA CTAA
+
+
+Since this is a riboswitch molecule, we may want to read it as ``RNA``.
+As the GenBank file usually have ``'t'`` instead of ``'u'`` in the
+sequence, we can not read it as ``RNA`` directly without conversion.
+We can read it as ``DNA`` and then ``transcribe`` it into ``RNA``:
+
+>>> rna_seq = dna_seq.transcribe()
```

Comment:
In better words: NCBI has a canonical form of `t` and we have a canonical form of `U`. Understanding that, we can safely make that conversion for the user. That way they are always operating in the canonical form of scikit-bio and don't have to worry that NCBI will use lowercase and `t`s instead of `U` in their file format. And when we write back out, we preserve this convention by changing all of our `U`s to `t`s so that we match NCBI.